ext/authx/Civi/Authx/Authenticator.php

```diff
@@ -128,7 +128,7 @@ public function auth($e, $details) {
     ]);
 
     if (isset($tgt->cred)) {
-      if ($principal = $this->checkCredential($tgt)) {
+      if ($principal = $this->checkCredential($tgt, implode('/', $e->args))) {
```

Comment:
I'm unclear what the $e is here...
- could it be typed? 
- would be nice if the docbloc shed a bit more light on it

Reply:
The docblock for `auth()` describes this as a `GenericHookEvent` corresponding to `civi.invoke.auth`. That event doesn't have much documentation, but the implementation is [pretty basic](https://github.com/civicrm/civicrm-core/blob/cdf7cc94efc5dd5433f43515945b60f684bb8f82/CRM/Core/Invoke.php#L62-L65).

--

As for why `$e` showed up in the test-failure -- there are actually two lines like this (one for `auth()` and one for `validate()`). The `auth()` one works with typical web-requests (via `civi.invoke.auth`), but`validate()` isn't part of that same event. Doh.

The real oversight here was that "requested HTTP path" (aka `$e->args`) isn't always a valid criterion for checking the credential. For example, these cases don't involve a well-defined path:

* Authenticating to a persistent HTTP session (*where the session has many different paths*)
* Authenticating to a [CLI pipe](https://docs.civicrm.org/dev/en/latest/framework/pipe/) (*with many actions and no HTTP routes*)
* Using Civi as a remote authn service -- eg there are existing callers who forward authentication-checks to APIv4 REST (`civicrm/ajax/api4/AuthxCredential/validate`) (*and they don't forward any meaningful path info*)

Obviously, none of these are actually intended to be used with the `PageToken` scenario. (And they won't be, because of the differing JWT scopes.) But the `CheckCredential` mechanism is shared, so `$requestPath` has to be set to *something* for all these cases.

So far, I'm finding it pretty workable to put in a `$requestPath` of `*`. (To wit: if a user tries to authenticate to a persistent session, then they're requesting access unrestricted/open-ended paths; `$requestPath` = `*`.)

Reply:
I added a WIP-commit for the idea of using `_aff` token to identity for the main page-view. (It moved a little bit to better respect module-boundaries, but still basically the same thing.) However, this technique is having an undesirable side-effect -- it causes the user-session to be authenticated. I'm not entirely sure why (and getting too sleep to dig into that). Hence the WIP-commit.

Reply:
Ooops, I think the last comment in this subthread was probably misplaced. I think we can close this subthread because the `$e` bits are addressed. I'll re-comment on the thread where that latter point matters.